src/lib/_internal_sense_hat.js

```diff
@@ -242,31 +242,43 @@ var $builtinmodule = function (name) {
         if (jsTemperature.value < -40 || jsTemperature.value > 120) {
             return Sk.ffi.remapToPy([0, jsTemperature.value]); // invalid
         }
-        
+
         return Sk.ffi.remapToPy([1, jsTemperature.value]);
     });
-    
+
     mod.fusionPoseRead = new Sk.builtin.func(function () {
         var fusionPose = Sk.ffi.remapToPy(Sk.sense_hat.rtimu.fusionPose);
-        
+
         return fusionPose;
     });
 
     mod.accelRead = new Sk.builtin.func(function () {
         var accel = Sk.ffi.remapToPy(Sk.sense_hat.rtimu.accel);
-        
+
         return accel;
     });
 
     mod.compassRead = new Sk.builtin.func(function () {
         var compass = Sk.ffi.remapToPy(Sk.sense_hat.rtimu.compass);
-        
+
         return compass;
     });
 
+    // Magnetometer tilt compensation
+    // From here http://diydrones.com/forum/topics/magnetometer-tilt-compensation-tutorial-sensor-analysis?commentId=705844%3AComment%3A1657670
+    mod.headingRead = new Sk.builtin.func(function () {
+        x = Sk.sense_hat.rtimu.fusionPose[0]
```

Comment:
You might add `var ` in front of every variable creation assignment, otherwise all those are global variables, which you might not want.